### PR TITLE
Add new device

### PR DESCRIPTION
**Pull request by MukeytheMuse**
Successfully adds the new device to the database; this can be tested through postman where you can authenticate using the device information (email/device id and password/device secret).
Screen Shot 2019-06-11 at 10 18 50 PM

### CODE REVIEW

**Comment by iamnukee** [comment]
nice! Could we get a screenshot of what it looks like please? (frontend)

File: src/app/components/create-new-device/create-new-device.component.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+      }
+
+      this.createDeviceForm = this.formBuilder.group({
+        name: new FormControl(name, [Validators.required]),
+        id: new FormControl(deviceId, [Validators.required]),
+        password: new FormControl(deviceSecret, [Validators.required]),
+      });
+
+      if (doUpdate) {
+        return;
```
Comment (iamnukee) [inline src/app/components/create-new-device/create-new-device.component.ts]: Is this needed?

Comment (MukeytheMuse) [reply src/app/components/create-new-device/create-new-device.component.ts]: Nope! Glad I forgot to remove that XD

**Comment by MukeytheMuse** [comment]
Screen Shot 2019-06-11 at 11 01 20 PM
Screen Shot 2019-06-11 at 11 01 28 PM
Screen Shot 2019-06-11 at 11 01 47 PM
Screen Shot 2019-06-11 at 11 01 56 PM

File: src/app/components/create-new-device/create-new-device.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+import { CreateNewDeviceComponent } from './create-new-device.component';
+
+describe('CreateNewDeviceComponent', () => {
+  let component: CreateNewDeviceComponent;
+  let fixture: ComponentFixture<CreateNewDeviceComponent>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [ CreateNewDeviceComponent ]
```
Comment (mll8657) [inline src/app/components/create-new-device/create-new-device.component.spec.ts]: if testing is something we still care about, this isn't going to work. This file will need things like the no errors schema and imports/providers for the classes it uses. login.component.spec.ts is a good example of what you need to do to get one of these testing files to work.

File: src/app/components/create-new-device/create-new-device.component.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  createDeviceForm: FormGroup;
+  submitted: boolean;
+  static PATH: any = 'devices/create';
+
+  constructor(private createDeviceService: DeviceService, private formBuilder: FormBuilder,
+              private route: ActivatedRoute, private router: Router, public dialog: MatDialog,
+              private snackBar: MatSnackBar) { }
+
+  ngOnInit() {
+    this.route.queryParams.subscribe(params => {
```
Comment (mll8657) [inline src/app/components/create-new-device/create-new-device.component.ts]: Move this to the constructor so it happens before the html loads.

Comment (MukeytheMuse) [reply src/app/components/create-new-device/create-new-device.component.ts]: All fixed with the newest commit.

**Comment by MukeytheMuse** [comment]
Definitely should have merged with master before committing! But no more conflicts and I updated the Path for the new device in the drop-down menu

**Pull request merged**